### ISSUE TITLE
Backup failed because: mkdir(): Permission denied

### ISSUE THREAD

**Issue by sixty8** [spatie/laravel-backup#611]
After latest 5.5 update i get this message: "Backup failed because: mkdir(): Permission denied",  in local (valet) and production server

**Comment by jpbalda** [comment]
Same happened to me today after upgrading :(

This is the notification I got in slack:

```
mkdir(): Permission denied

#0 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(2, 'mkdir(): Permis...', '/var/www/app...', 44, Array)
#1 /var/www/app/vendor/spatie/temporary-directory/src/TemporaryDirectory.php(44): mkdir('/var/www/app...', 511, true)
#2 /var/www/app/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php(121): Spatie\TemporaryDirectory\TemporaryDirectory->create()
#3 /var/www/app/vendor/spatie/laravel-backup/src/Commands/BackupCommand.php(49): Spatie\Backup\Tasks\Backup\BackupJob->run()
#4 [internal function]: Spatie\Backup\Commands\BackupCommand->handle()
#5 /var/www/app/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array(Array, Array)
#6 /var/www/app/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#7 /var/www/app/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure))
#8 /var/www/app/vendor/laravel/framework/src/Illuminate/Container/Container.php(549): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL)
#9 /var/www/app/vendor/laravel/framework/src/Illuminate/Console/Command.php(183): Illuminate\Container\Container->call(Array)
#10 /var/www/app/vendor/symfony/console/Command/Command.php(252): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#11 /var/www/app/vendor/laravel/framework/src/Illuminate/Console/Command.php(170): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle))
#12 /var/www/app/vendor/spatie/laravel-backup/src/Commands/BaseCommand.php(22): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /var/www/app/vendor/symfony/console/Application.php(938): Spatie\Backup\Commands\BaseCommand->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /var/www/app/vendor/symfony/console/Application.php(240): Symfony\Component\Console\Application->doRunCommand(Object(Spatie\Backup\Commands\BackupCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /var/www/app/vendor/symfony/console/Application.php(148): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#16 /var/www/app/vendor/laravel/framework/src/Illuminate/Console/Application.php(88): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#17 /var/www/app/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(121): Illuminate\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#18 /var/www/app/artisan(37): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#19 {main}
```

**Comment by jpbalda** [comment]
Found the problem...
Since I have a cron job running every night to execute the task, the backup was executed successfully yesterday for the first time after the upgrade and the folder .../my-app/storage/app/backup-temp was created with owner and group root. That's why when I ran the task manually logged in as my deploy user, the task failed.

To fix it, I only changed the folder's owner and group to my deploy user.

Hope that helps :)
